src/helpers.js

```diff
@@ -5,10 +5,16 @@ const TEXT_NODE = 3
 
 // Currently this fn only supports jest timers, but it could support other test runners in the future.
 function runWithRealTimers(callback) {
-  return _runWithRealTimers(callback).callbackReturnValue
+  // istanbul ignore else
+  if (typeof jest !== 'undefined') {
```

Comment:
note that https://jestjs.io/docs/en/configuration#injectglobals-boolean is a thing, so if people enable that this check will give a false positive.

Not sure what the best solution is...

Reply:
Possibly we could stick some non+enumerable symbol on the global object?

Reply:
Thanks for the pointer!

I'm honestly at the point where we should just create some wrapper library that requires passing some adapter to fake/real timers. That logic is just way too hairy for my liking and we have to consider quite a number of environments by now that we don't actually test with.

Using the `exports` field might be the best approach to stick with zero-config but I don't know if we could/should add an extra `jest` or `mocha` or `browser` entry.

PS: [`react` has the same check](https://github.com/eps1lon/react/blob/09916479219a61ae86d2ec8ce159a161337b9007/packages/react-reconciler/src/ReactFiberHooks.old.js#L1419-L1422) so ideally we fix both instances so that we don't create even more pseudo-contexts (testing-library/react vs react in jest without globals).

Reply:
I came over this since I had to revert the `@testing-library/dom` update as it broke our tests using fake timers 😅 

Not sure if Jest should put somewhere "I'm here, running code and stuff" or if libraries should take config. I like the former as a consumer but it requires you to keep detection (albeit a "blessed" one), and creating adaptors for everything doesn't sound scalable. And makes it potentially harder for other libraries to integrate cleanly... Dunno what's best

Reply:
> I came over this since I had to revert the @testing-library/dom update as it broke our tests using fake timers sweat_smile

Which version did you have to rollback?

Reply:
Is there any way jest exposes for us to detect whether we're running in a jest environment? I think that this feature is very useful and considering the vast majority of users are using jest it seems appropriate to include even as a "blessed" solution. I think it would make sense to support other popular runners as well assuming it doesn't complicate the codebase too much.

Is there an environment variable or something we can use to reliably detect whether we're running in a jest environment?

Reply:
> > I came over this since I had to revert the @testing-library/dom update as it broke our tests using fake timers sweat_smile
> 
> Which version did you have to rollback?

We were (and are back to) using `7.29.4`. Got `7.29.6` when updating, did not try `.5`

> Is there an environment variable or something we can use to reliably detect whether we're running in a jest environment?

Not as of now, that's what I was suggesting with the `Symbol`. I guess we could put something on `process.env`, tho, that sounds like a good idea. That might make sense regardless even though it requires polyfilling `process.env` for browser usage to use it. Jest won't be the _runner_ in the browser though (at least not any time soon), so I guess that's not really a concern

Reply:
We have `JEST_WORKER_ID`, maybe you can use that?

https://jestjs.io/docs/environment-variables#jest_worker_id